src/main/java/org/scalameter/japi/annotation/benchmark.java

```diff
@@ -0,0 +1,31 @@
+package org.scalameter.japi.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+
+/** Annotates benchmark snippet.
+ *
+ *  Its value sets the scope.
+ *  {{
+ *    @benchmark("arrays.forloops")
+ *
+ *    // example Scala DSL equivalent
+ *    performance of "arrays" in {
+ *      measure method "forloops" in {
+ *
+ *      }
+ *    }
+ *  }}
+ *
+ *  Note that method annotated with `@benchmark` should be public
+ *  and have single argument that is a type of a generator
+ *  that is found by required `@gen` annotation.
```

Comment:
... and have a single argument of type T, such that it corresponds to the
    type of the generator declared in the @gen annotation.